libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -301,16 +306,25 @@ AP_GPS::detect_instance(uint8_t instance)
     state[instance].status = NO_GPS;
     state[instance].hdop = 9999;
 
+    switch (_type[instance]) {
 	// by default the sbf/trimble gps outputs no data on its port, until configured.
-	if (_type[instance] == GPS_TYPE_SBF) {
+    case GPS_TYPE_SBF:
 		_broadcast_gps_type("SBF", instance, -1); // baud rate isn't valid
 		new_gps = new AP_GPS_SBF(*this, state[instance], _port[instance]);
-	} else if ((_type[instance] == GPS_TYPE_GSOF)) {
+        break;
+
+    case GPS_TYPE_GSOF:
 		_broadcast_gps_type("GSOF", instance, -1); // baud rate isn't valid
 		new_gps = new AP_GPS_GSOF(*this, state[instance], _port[instance]);
-	} else if ((_type[instance] == GPS_TYPE_NOVA)) {
+        break;
+
+    case GPS_TYPE_NOVA:
 		_broadcast_gps_type("NOVA", instance, -1); // baud rate isn't valid
 		new_gps = new AP_GPS_NOVA(*this, state[instance], _port[instance]);
+        break;
```

Comment:
All the breaks aren't aligned with previous lines (because those lines have tabs instead of spaces).